### PR TITLE
Switch from Nose to Pytest Runner

### PR DESCRIPTION
**Pull request by SethMichaelLarson**
The first step in the effort to switch to `py.test` as opposed to `nose`/`unittest` (#1160).
So far I've hit two problems: 
- Apparently `py.test` reloads the entire module per-test and emits the "No true SSLContext object available" warning for each test which luckily only affects Python 2.6 `test_https.TestHTTPS.test_ssl_wrong_system_time` and `test_https.TestHTTPS.test_ssl_correct_system_time` tests. Id' advocate just skipping these tests on Python versions without a correct SSLContext.

- Google App Engine, I don't know much about it or how testbeds work so I could use some direction here.

### CODE REVIEW

File: test/appengine/nose.cfg [hunk 1]
```diff
@@ -1,4 +0,0 @@
-[nosetests]
-cover-min-percentage=0
-with-gae=1
-gae-application=test/appengine/app.yaml
```
Comment (Lukasa) [inline test/appengine/nose.cfg]: Yeah, these look like things that will require manual intervention to reproduce.

File: test/contrib/test_gae_manager.py [hunk 2]
```diff
@@ -57,9 +61,6 @@ def _absolute_url(self, path):
 class TestGAEConnectionManager(TestConnectionPool):
     __test__ = True
 
-    # Magic class variable that tells NoseGAE to enable the URLFetch stub.
-    nosegae_urlfetch = True
```
Comment (Lukasa) [inline test/contrib/test_gae_manager.py]: Do we need an equivalent to this for pytest_gae?

Comment (SethMichaelLarson) [inline test/contrib/test_gae_manager.py]: I'm reading the docs as we speak. :)

Comment (SethMichaelLarson) [inline test/contrib/test_gae_manager.py]: Okay I think I'm going to switch from pytest_gae which looks like a unmaintained project (?) to pytest_beds which at least mentions urlfetch in it's docs. Will do some more research.

**Comment by SethMichaelLarson** [comment]
I've switched from `warnings.catch_warnings()` to using `pytest.warns(None)`, this might resolve our issue with Python 2.6 but we shall see.

**Comment by SethMichaelLarson** [comment]
Nope! Python 2.6 still throwing that SSLContext warning. I'll skip on those Python versions.

**Comment by SethMichaelLarson** [comment]
Hitting one of the pain points of `py.test`-style tests with GAE tests. Fixtures and test inheritance don't mix well. This is the only thing I dislike about `py.test` compared to `unittest`/`nose`. :cry:

**Comment by SethMichaelLarson** [comment]
Alright I'm stuck for now, I don't have an AppEngine environment to manually track down why it's not calling `google.appengine.api.urlfetch.fetch` in `TestHTTP.test_urlfetch_called_with_http`. I think this is the only issue that's remaining for this PR.

**Comment by SethMichaelLarson** [comment]
Pretty sure you didn't mean to approve these changes @Lukasa, would be a train wreck to merge now. :wink:

**Comment by SethMichaelLarson** [comment]
I don't have enough time to get this PR 100% ready to merge as I'm stuck on GAE and don't have time to create my own local environment to test things.  Seems as though @alexwlchan is putting this work in so I'll close this.

**Pull request closed without merging**